addons/misra.py

```diff
@@ -1461,6 +1461,79 @@ def reportErrorIfVariableIsNotConst(variable, stringLiteral):
                             reportErrorIfVariableIsNotConst(parameterDefinition.nameToken, usedParameter)
 
+    def misra_8_2(self, data, rawTokens):
+        def getFollowingRawTokens(rawTokens, token, count):
+            following =[]
+            for rawToken in rawTokens:
+                if (rawToken.file == token.file and
+                        rawToken.linenr == token.linenr and
+                        rawToken.column == token.column):
+                    for _ in range(count):
+                        rawToken = rawToken.next
+                        # Skip comments
+                        while rawTokens and (rawToken.str.startswith('/*') or rawToken.str.startswith('//')):
+                            rawToken = rawToken.next
+                        if rawToken is None:
+                            break
+                        following.append(rawToken)
+
+                    return following
+
+        # Check arguments in function declaration
+        for func in data.functions:
+
+            # Zero arguments should be in form ( void )
+            if (len(func.argument) == 0):
+                zeroCallTokens = getFollowingRawTokens(rawTokens, func.tokenDef, 3)
+                if len(zeroCallTokens) != 3:
+                    continue
+                if (zeroCallTokens[0].str != '(' or
+                        zeroCallTokens[1].str != 'void' or
+                        zeroCallTokens[2].str != ')'):
+                    self.reportError(func.tokenDef, 8, 2)
+
+            startCall = func.tokenDef.next
+            if startCall is None or startCall.str != '(':
+                continue
+
+            endCall = startCall.link
+            if endCall is None or endCall.str != ')':
+                continue
+
+            for arg in func.argument:
+                argument = func.argument[arg]
+                typeStartToken = argument.typeStartToken
+                if typeStartToken is None:
+                    continue
+
+                nameToken = argument.nameToken
+                # Arguments should have a name unless variable length arg
+                if nameToken is None and typeStartToken.str != '...':
+                    self.reportError(typeStartToken, 8, 2)
+
+                # Type declaration on next line (old style declaration list) is not allowed
+                if (typeStartToken.linenr > endCall.linenr) or (typeStartToken.column > endCall.column):
+                    self.reportError(typeStartToken, 8, 2)
+
+        # Check arguments in pointer declarations
+        for var in data.variables:
+            if not var.isPointer:
+                continue
+
+            if var.nameToken is None:
+                continue
+
+            rawTokensFollowingPtr = getFollowingRawTokens(rawTokens, var.nameToken, 3)
+            if len(rawTokensFollowingPtr) != 3:
+                continue
+
+            # Compliant:           returnType (*ptrName) ( ArgType )
```

Comment:
I have the feeling that the long term goal would be that the tokenlist has all this information.. but maybe for now it will be easier that you check the rawtokens like this..